src/Dialect/ONNX/ONNXOpsHelper.cpp

```diff
@@ -12,11 +12,13 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include "src/Dialect/ONNX/ONNXOpsHelper.hpp"
 #include "mlir/IR/TypeUtilities.h"
-
-#include "src/Dialect/ONNX/IndexExpr.hpp"
+#include "src/Dialect/Krnl/KrnlHelper.hpp"
```

Comment:
That is what you don't want. We "must" keep ONNX Dialect free of anything KRNL related. No refs to any header files. This will forces users of ONNX Dialects that are only interested in ONNX to have krnl, which they have requested not to have.

Reply:
OK I will think about how to reorg the code to minimize dependencies. Might have questions later.

Reply:
I have changed the patch to remove any change to the shared libraries structure. To do that I have created a class  in ONNXToKrnlCommon containing the member functions I need to lower both `SpaceToDepth` and `DepthToSpace` operators. The actual code in those member functions is unchanged. Please take another look @AlexandreEichenberger.

Reply:
LGTM, thanks for preserving the independence of ONNX vs KRNL, much appreciated